src/SourceWatcherFuncApp/Utilities/EntityHelper.cs

```diff
@@ -31,7 +30,7 @@ public static DetectorEntity PrepareEntityForLoad(Stream streamAssemblyData, str
                 detectorPackage.IsInternal = resourceFilter != null? resourceFilter.InternalOnly : false;
                 detectorPackage.SupportTopicList = invoker.EntryPointDefinitionAttribute != null ? invoker.EntryPointDefinitionAttribute.SupportTopicList : new List<SupportTopic>() ;
                 detectorPackage.AnalysisTypes = invoker.EntryPointDefinitionAttribute != null ? invoker.EntryPointDefinitionAttribute.AnalysisTypes : new List<string>();
-
+                detectorPackage.DetectorType = invoker.EntryPointDefinitionAttribute != null ? invoker.EntryPointDefinitionAttribute.Type.ToString() : "Detector";
```

Comment:
instead of static string of "Detector" you can do `default(DetectorType)`

Reply:
Good point

Reply:
But how do we know the default of `DetectorType `is "Detector"?